lib/primitives/mtx.js

```diff
@@ -158,11 +164,19 @@ class MTX extends TX {
   addCoin(coin) {
     assert(coin instanceof Coin, 'Cannot add non-coin.');
 
-    const input = Input.fromCoin(coin);
+    // Avoid duplicate inputs
+    for (const input of this.inputs) {
```

Comment:
Another note is this is inefficient way of checking this `O(n^2)`. If we really wanted to do this(and as I mentioned above, we should not), we have to create new Map for the hash/index pair of each input similar to CoinView (Not coinview is not good case for this, as it may not have some coins)